### PR TITLE
(CDAP-13135) Remove some usages of the deprecated InputSupplier and OutputSupplier

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationAdapter.java [hunk 1]
```diff
@@ -51,7 +51,6 @@
 import com.google.common.base.Preconditions;
 import com.google.common.base.Throwables;
 import com.google.common.collect.Maps;
-import com.google.common.io.InputSupplier;
 import com.google.common.io.OutputSupplier;
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationAdapter.java]: can this be removed?

File: cdap-common/src/test/java/co/cask/cdap/common/lang/jar/BundleJarUtilTest.java [hunk 2]
```diff
@@ -46,6 +44,14 @@
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  @Test
+  public void test() throws Exception {
```
Comment (albertshau) [inline cdap-common/src/test/java/co/cask/cdap/common/lang/jar/BundleJarUtilTest.java]: was this test case intended? Looks like a sanity check

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationAdapter.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationAdapter.java]: Oh yeah. Optimizing the imports.

File: cdap-common/src/test/java/co/cask/cdap/common/lang/jar/BundleJarUtilTest.java (hunk 2 above)
Comment (chtyim) [reply cdap-common/src/test/java/co/cask/cdap/common/lang/jar/BundleJarUtilTest.java]: Oops.. forget to remove.

**Comment by chtyim** [comment]
@albertshau Addressed comments.

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6667-6

**Comment by chtyim** [comment]
Build passed, squash and merge.

**Pull request merged**